data/tutorials/language/0lg_08_env.md

```diff
@@ -0,0 +1,557 @@
+```ocaml
+# ( + ) 2;;
+- : int -> int = <fun>
+```
+
+Passing a single integer to the addition returns a function of type `int -> int`. That function value is a closure; the value passed as the first parameter, here 2, is captured as if it had been in an earlier definition.
+
+#### Passing Multiple Arguments as Tuples
+
+Consider this function:
+```ocaml
+# let space_cat (s, t) = s ^ " " ^ t;;
+val space_cat : string * string -> string
+```
+
+It takes a pair of strings and returns a string.
+```ocaml
+# space_cat ("hello", "world");;
+- : string = "hello world"
+```
+
+Asking whether `space_cat` is a binary function makes sense. Morally, it takes two parameters: the strings that are sandwiching the space character. Syntactically, the call `space_cat ("hello", "world")` looks pretty much like a function call in high school maths or in a spreadsheet application.
+
+However, from an OCaml perspective, `space_cat` takes a single parameter, which is a pair.
+
+### Currying and Uncurrying
+
+In the two previous sections, two kinds of “multiple parameter” functions have been presented.
+- Function returning a function, such as `( + )`
+- Functions taking a pair or a tuple as a parameter, such as `space_cat`
+
+Interestingly, both kinds of functions provide a way to pass several parameters while being functions with a single parameter. In this perspective, it makes sense to say: “All functions have a single argument”.
+
+This goes even further. Any function of the same kind can be translated into an equivalent function of the second kind, and conversely. Using OCaml support for higher-order functions, it is possible to define those transformations as functions.
+
+Here is the translation from the first kind to the second kind:
+```ocaml
+# fun f -> fun (x, y) -> f x y;;
+- : ('a -> 'b -> 'c) -> 'a * 'b -> 'c = <fun>
+```
+
+Here is the reverse translation:
+```ocaml
+# fun f -> fun x y -> f (x, y);;
+- : ('a * 'b -> 'c) -> 'a -> 'b -> 'c = <fun>
+```
+
+These translations are attributed to the 20th-century logician [Haskell Curry](https://en.wikipedia.org/wiki/Haskell_Curry). The second translation is called currying and the first is called uncurrying.
+
+From a typing perspective, this means that for any types `a`, `b`, and `c`, the following types are equivalent:
+- `a -> (b -> c)` &mdash; curried function type
+- `a * b -> c` &mdash; uncurried function type
+
+We will not dive any deeper into the details. This equivalence can be formally defined using _ad-hoc_ mathematics.
+
+It is very rare to need to use the functions above. However, it is important to understand that changing a function may not need any code refactoring. A function can be given several equivalent forms and changed into another, either using refactoring or using a higher-order function. Since functions are values, currying and uncurrying are operations on those values.
```

Comment:
```suggestion
It is rarely necessary to use the functions above. However, it is important to understand that changing a function may not need any code refactoring. A function can be given several equivalent forms and changed into another, either using refactoring or using a higher-order function. Since functions are values, currying and uncurrying are operations on those values.
```